MailCore/Cache/MailboxManager/MailboxManager+Sentry.swift

```diff
@@ -0,0 +1,62 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2023 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import Sentry
+
+// MARK: - Sentry
+
+public extension MailboxManager {
+    enum Category: String {
+        case APIError = "MailboxManagerAPIError"
+        case RealmError = "MailboxManagerRealmError"
+    }
+
+    /// Observe API callback errors to generate a dashboard  on sentry
+    @discardableResult
+    func observeAPIErrors<T>(_ function: String = #function, _ task: @escaping () async throws -> T) async throws -> T {
```

Comment:
This works by wraping whatever call we want to observe errors passing by. 
It returns or throws what is wraped in `task`

Reply:
Why is it easier like this instead of doing it in `perform` here `MailApiFetcher.swift:46` ?

Reply:
This version is not easier, I have the context of who is calling availlable tho. Yeah too much fluf. I'll change to `MailApiFetcher.swift:46`